common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java

```diff
@@ -0,0 +1,834 @@
+                                              @NonNull final URL requestUrl,
+                                              @NonNull final String accessToken,
+                                              @Nullable final String nonce) throws ClientException {
+        final Exception exception;
+        final String errCode;
+
+        try {
+            final JWTClaimsSet.Builder claimsBuilder = new JWTClaimsSet.Builder();
+            claimsBuilder.claim(SignedHttpRequestJwtClaims.ACCESS_TOKEN, accessToken);
+            claimsBuilder.claim(SignedHttpRequestJwtClaims.TIMESTAMP, System.currentTimeMillis() / 1000L);
+            claimsBuilder.claim(SignedHttpRequestJwtClaims.HTTP_METHOD, httpMethod);
+            claimsBuilder.claim(SignedHttpRequestJwtClaims.HTTP_HOST, requestUrl.getHost());
+            claimsBuilder.claim(SignedHttpRequestJwtClaims.HTTP_PATH, requestUrl.getPath());
+            claimsBuilder.claim(SignedHttpRequestJwtClaims.CNF, getDevicePopJwkMinifiedJson());
+            claimsBuilder.claim(SignedHttpRequestJwtClaims.NONCE, nonce);
+
+            final JWTClaimsSet claimsSet = claimsBuilder.build();
+
+            final KeyStore.Entry entry = mKeyStore.getEntry(KEYSTORE_ENTRY_ALIAS, null);
+            final PrivateKey privateKey = ((KeyStore.PrivateKeyEntry) entry).getPrivateKey();
+            final RSASSASigner signer = new RSASSASigner(privateKey);
+
+            final SignedJWT signedJWT = new SignedJWT(
+                    new JWSHeader.Builder(JWSAlgorithm.RS256)
+                            .build(),
+                    claimsSet
+            );
+
+            signedJWT.sign(signer);
+
+            return signedJWT.serialize();
+        } catch (final NoSuchAlgorithmException e) {
+            exception = e;
+            errCode = NO_SUCH_ALGORITHM;
+        } catch (final KeyStoreException e) {
+            exception = e;
+            errCode = KEYSTORE_NOT_INITIALIZED;
+        } catch (final JOSEException e) {
+            exception = e;
+            errCode = JWT_SIGNING_FAILURE;
+        } catch (final UnrecoverableEntryException e) {
+            exception = e;
+            errCode = INVALID_PROTECTION_PARAMS;
+        }
+
+        final ClientException clientException = new ClientException(
+                errCode,
+                exception.getMessage(),
+                exception
+        );
+
+        Logger.error(
+                TAG,
+                clientException.getMessage(),
+                clientException
+        );
+
+        throw clientException;
+    }
+
+    //region Internal Functions
+
+    /**
+     * Generates a new RSA KeyPair of the specified lenth.
+     *
+     * @param ctx        The current application Context.
+     * @param minKeySize The minimum keysize to use.
+     * @return The newly generated RSA KeyPair.
+     * @throws UnsupportedOperationException
+     */
+    @SuppressLint("NewApi")
+    private KeyPair generateNewRsaKeyPair(@NonNull final Context ctx,
```

Comment:
nit: Rename `ctx` to `context`

Reply:
Resolved in 2f3598f4708f8ef7a85313ab85819260d9eb185a